### PR TITLE
Add China holidays definition.

### PR DESCRIPTION
**Pull request by Eric-Guo**
East Asia country like China, Korea and Japan using the [Lunisolar calendar](https://en.wikipedia.org/wiki/Lunisolar_calendar) to define many traditional festival, thanks @sunsidew [work](https://github.com/sunsidew/ruby_lunardate), make it possible to implement relative easier in Ruby ecosystem.

Currently those holidays still have below issue, but due to my limited knowledge about astronomy, I have to leave as issue:

* I using jp_vernal_equinox_day, but seems there is little different, `2005-4-5` is correct 清明 in China, according to the history.
* I check several days of `ruby_lunardate`, seems all result are right in Chinese Calendar, but still can not confirm, in face I not known how and where the `KOREAN_LUNAR_YEAR_INFO` coming from.

Anyway, it's a small forward for a worldwide holidays gems. :smile:

### CODE REVIEW

**Comment by ptrimble** [comment]
This is fantastic! I will definitely look this over but on the face of it I think it looks great!

Now, have we talked with @sunsidew about using his code? I know that we *can* but I would prefer that we make sure s/he is on board and has no issues with the copy.

File: lib/holidays/date_calculator/lunisolar.rb [hunk 1]
```diff
@@ -0,0 +1,218 @@
+module Holidays
+  module DateCalculator
+    # Copied from Korean Lunar/Solar Calendar Library, by sunsidew
+    # https://github.com/sunsidew/ruby_lunardate
```
Comment (Eric-Guo) [inline lib/holidays/date_calculator/lunisolar.rb]: @sunsidew, I'm not know if it's already enough to for [BSD](https://github.com/sunsidew/ruby_lunardate/blob/master/ruby_lunardate.gemspec#L12) require?

Comment (sunsidew) [inline lib/holidays/date_calculator/lunisolar.rb]: @ptrimble @Eric-Guo Sure. If there's one thing which I wish, [ruby_lunardate](https://github.com/sunsidew/ruby_lunardate) repository will be mentioned in credit. Is it okay?

**Comment by sunsidew** [comment]
@Eric-Guo Chinese lunar date isn't same as Korean. So you will be change some data in `KOREAN_LUNAR_YEAR_INFO`. [#6](https://github.com/sunsidew/ruby_lunardate/issues/6) will help you.

File: lib/holidays/date_calculator/lunisolar.rb (hunk 1 above)
Comment (ptrimble) [inline lib/holidays/date_calculator/lunisolar.rb]: Yes, of course. I'll make sure I handle this when I perform the updates you listed below. Thank you!

**Comment by ptrimble** [comment]
@Eric-Guo Are you willing to make the necessary update for the `KOREAN_LUNAR_YEAR_INFO` as mentioned by @sunsidew? Or would you prefer that I finish it up and manage the conflicts (attributing the commits to you, of course)? Your call. Please let me know.

**Comment by Eric-Guo** [comment]
Kindly give me sometime to finalize the PR, I plan to add CHINESE_LUNAR_YEAR_INFO via another data source, but need write a program to convert to such data.

I will also handle the conflicts of the master branch of cause.

**Comment by ptrimble** [comment]
Fantastic, thank you! I'll wait to hear back.

**Comment by ptrimble** [comment]
@Eric-Guo I've changed my thinking just a bit based on the work I've done this weekend on my refactor branch. Let me lay out my thoughts and you can let me know what you think about the timeline and how reasonable (or unreasonable!) it is.

My current plan was to include this PR in a `3.3.0` release so that it was available before my `4.0.0` breaking change. This way users could receive the benefit of your work but not need to worry about the breaking changes contained in my release. However, I don't want to let my large refactor drift too much. It's been sitting out for 10 days now. So with that in mind I'm going to set an entirely arbitrary deadline for your PR of **Sunday, 4/17**.

There is absolutely nothing concrete about this date. It is totally open for negotiation. If you think you'll need more time or if you think this date is unreasonable then let's talk! 

We have some options:
* push this arbitrary date back a bit so we can get this PR in with `3.3.0`
* I can merge my changes in and you can release this PR in a future point release (like `4.1.0`). The downside to this one is that you'll be required to manage more merge conflicts. The refactor was pretty big and while I personally think it is manageable it will be more work for you. 😦 

Let me know what you think! Again, no need to stress, we can work something out that works for both of us.

**Comment by ptrimble** [comment]
Hello @Eric-Guo! I'm dreading another flood of PRs that will require significant work for me to rebase. So as I said in a previous comment I'm going to merge in my breaking change tomorrow if I don't hear from you. Then I will rebase your branch and provide you with an 'updated' version that you can use to complete your changes. I honestly don't think it will be too bad.

I know that it will be a bit more painful but I really do want to get my refactor out. Hope you understand. 😄 

Thanks!

**Comment by Eric-Guo** [comment]
You can leave this PR open and even no need todo rebase to save time, I will checking your change and do a update latter.

I plan to using lunar calculate from [another library](https://github.com/isee15/Lunar-Solar-Calendar-Converter), so spent your valuable time on current status is wasting.

Thanks!

**Comment by ptrimble** [comment]
Okay! Thanks for understanding! I will be merging my refactor in the next 24 hours, then. Please feel free to contact me if you have any questions while rebasing.

**Comment by ptrimble** [comment]
Hello @Eric-Guo! Just checking in. I would be more than happy to take this over and make necessary changes to fix the merge conflicts if you don't have the time right now to do so, especially since the entire reason for the conflicts was a big refactor on my part. 😄 

Let me know!

**Comment by Eric-Guo** [comment]
@ptrimble You are so kind! I just change the job, and new job generate higher amount of works to me at least for a while...

But the main problem of this PR is according to the Chinese Holiday, based on [Korea Lunar still not accurate](https://github.com/sunsidew/ruby_lunardate/issues/6), so simple reflect won't fix that and release a wrong implementation are worse than no implementation.

I just close this PR and I will give out another correct PR when I got enough tim, for any one interested in holiday in China also, he can reference this PR, but must using Lunar calendar based on (Lunar-Solar-Calendar-Converter)[https://github.com/isee15/Lunar-Solar-Calendar-Converter] instead of Korea.

**Comment by ptrimble** [comment]
Thanks so much @Eric-Guo! I am going to leave it as-is since I know basically nothing about Chinese holidays and, as you so rightly said, it is worse to release a wrong implementation than no implementation.

Thanks again for taking the time to investigate this! Best of luck in your new job.

**Pull request closed without merging**